cmd/ci-operator/main_test.go

```diff
@@ -505,3 +506,49 @@ func TestLoadLeaseCredentials(t *testing.T) {
 		})
 	}
 }
+
+func TestExcludeContextCancelledErrors(t *testing.T) {
+	testCases := []struct {
+		id       string
+		errs     []error
+		expected []error
+	}{
+		{
+			id: "no context cancelled errors, no changes expected",
+			errs: []error{
+				results.ForReason("step_failed").WithError(errors.New("oopsie")).Errorf("step foo failed: oopsie"),
+				results.ForReason("step_failed").WithError(errors.New("oopsie")).Errorf("step bar failed: oopsie"),
+			},
+			expected: []error{
+				results.ForReason("step_failed").WithError(errors.New("oopsie")).Errorf("step foo failed: oopsie"),
+				results.ForReason("step_failed").WithError(errors.New("oopsie")).Errorf("step bar failed: oopsie"),
+			},
+		},
+		{
+			id: "context cancelled errors, changes expected",
+			errs: []error{
+				results.ForReason("step_failed").WithError(errors.New("oopsie")).Errorf("step foo failed: oopsie"),
+				results.ForReason("step_failed").WithError(context.Canceled).Errorf("step bar failed: %v", context.Canceled),
+			},
+			expected: []error{
+				results.ForReason("step_failed").WithError(errors.New("oopsie")).Errorf("step foo failed: oopsie"),
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		actualErrs := excludeContextCancelledErrors(tc.errs)
+
+		printErrors := func(errs []error) string {
+			var ret string
+			for _, e := range errs {
+				ret += e.Error() + "\n"
+			}
+			return ret
+		}
+
+		if !reflect.DeepEqual(actualErrs, tc.expected) {
```

Comment:
I spend an hour trying to figure out how this can work with `cmp.Diff`. I tried almost everything with `AllowUnexported` but I gave up.

Reply:
```go
if diff := cmp.Diff(actualErrs, tc.expected, testhelper.EquateErrorMessage); diff != "" {
	t.Fatal(diff)
}
```
seems to work.

Reply:
Thanks. updated